compiler/next/lib/queries/Context.cpp

```diff
@@ -72,7 +97,10 @@ Context::~Context() {
   for (auto& item: uniqueStringsTable) {
     char* buf = (char*) item.str;
     buf -= UNIQUED_STRING_METADATA_BYTES;
-    free(buf);
+    char doNotCollect = buf[UNIQUED_STRING_METADATA_LEN + 1];
```

Comment:
Shouldn't this code be freeing the buf if `this == &rootContext` so we don't have any memory leak at all?